lib/std/process.zig

```diff
@@ -299,196 +301,279 @@ pub const ArgIteratorWasi = struct {
     }
 };
 
-pub const ArgIteratorWindows = struct {
-    index: usize,
-    cmd_line: [*]const u16,
+/// Optional parameters for `ArgIteratorGeneral`
+pub const ArgIteratorGeneralOptions = struct {
+    comments_supported: bool = false,
+};
 
-    pub const NextError = error{ OutOfMemory, InvalidCmdLine };
+/// A general Iterator to parse a string into a set of arguments
+pub fn ArgIteratorGeneral(comptime options: ArgIteratorGeneralOptions) type {
+    return struct {
+        allocator: Allocator,
+        index: usize = 0,
+        cmd_line: [:0]const u8,
+
+        start: usize = 0,
+        end: usize = 0,
+        buffer: []u8,
+
+        pub const Self = @This();
+
+        pub const InitError = error{OutOfMemory};
+        pub const InitUtf16leError = error{ OutOfMemory, InvalidCmdLine };
+
+        /// cmd_line_utf8 is duplicated into an internal buffer
+        pub fn init(allocator: Allocator, cmd_line_utf8: []const u8) InitError!Self {
```

Comment:
why is it duplicated into an internal buffer? seems unnecessary. certainly it is unnecessary for the one use case that we have for it.

Reply:
This is duplicated because the other init*() functions needed to duplicate, and I was just trying to keep it simple.

But it's not too difficult to add an optional member that holds the cmd_line if it should be free'd by the deinit().